### PR TITLE
Add dijkstra algorithm

### PR DESCRIPTION
**Pull request by kapildeshpande**

### CODE REVIEW

**Comment by faheel** [comment]
There have been lots of suggestions recently by interested contributors to add more algorithms and data structures to the project, which I have currently put on hold because I want to close issue #74 first, as it would require a complete reorganisation of files and some refactoring of code.

I previously thought that I would work on #74 once the currently active PRs are merged, but I feel that it would take much longer that way. So, if it's not too inconvenient for you, I would like to work on issue #74 as soon as possible. The remaining work on this PR (and others) can be carried on afterwards. Once the project is refactored, all you would need to do is `git fetch upstream` and move the code that you wrote to the new appropriate directory.

So, what do you say?

**Comment by kapildeshpande** [comment]
Ok no problem.

File: C++/source/Graph algorithms/Dijkstra_algorithm.cpp
Comment (faheel) [inline C++/source/Graph algorithms/Dijkstra_algorithm.cpp]: Please make these containers local to `main()` and pass them by reference to whichever functions need them.

Also, you can declare `visited` and `dist` as `vector<bool>` and `vector<int>` respectively, which would remove the need for using the constant `MAX`.

File: C++/source/Graph algorithms/Dijkstra_algorithm.cpp
Comment (faheel) [inline C++/source/Graph algorithms/Dijkstra_algorithm.cpp]: Use `num_` instead of `no_of_`, since it is a more widely used convention.

**Review by faheel — CHANGES_REQUESTED** [review]
Thanks for adding Dijkstra's algorithm; it has been long overdue :sweat_smile:. It would be great if you could also add unit tests for it with a few test cases. The details for it can be found in [UNIT_TESTS.md](https://github.com/faheel/Algos/blob/master/C%2B%2B/UNIT_TESTS.md).

Also, please rename the file to just `dijkstra.cpp` and its parent directory to `graph`.

**Comment by kapildeshpande** [comment]
Ok no problem

**Comment by faheel** [comment]
Closing this due to lack of activity for a long time.

**Pull request closed without merging**
